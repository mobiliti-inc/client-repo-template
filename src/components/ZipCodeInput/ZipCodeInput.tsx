import React from 'react';
import TextInput, { INPUT_TYPES } from '../TextInput/TextInput';

import styles from './ZipCodeInput.scss';

class ZipCodeInput extends React.PureComponent {
	static propTypes = {
		value: PropTypes.string.isRequired,
		valid: PropTypes.bool.isRequired,
		onChange: PropTypes.func.isRequired,
	}

	validate = ({ target: { value } }) => {
		let valid = true;
		if (!/^\d{5}(?:[-\s]\d{4})?$/.test(value)) valid = false;
		const zipCode = {
			value,
			valid,
		};
		this.props.onChange(zipCode);
	}

	render() {
		return (
			<TextInput
				id="input-zip-code"
				type={INPUT_TYPES.TEXT}
				placeholder="Zip Code"
				value={this.props.value}
				isValid={this.props.valid}
				onChange={this.validate}
				dark
			/>);
	}
}

export default CSSModules(ZipCodeInput, styles, { allowMultiple: true });
