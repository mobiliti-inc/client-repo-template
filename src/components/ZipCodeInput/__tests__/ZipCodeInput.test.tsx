import ZipCodeInput from '../ZipCodeInput';

describe('ZipCodeInput', () => {
	it('renders', () => {
		const subject = shallow(<ZipCodeInput
			value=""
			onChange={() => {}}
			valid
		/>);
		matchSnapshot(subject);
	});
});
