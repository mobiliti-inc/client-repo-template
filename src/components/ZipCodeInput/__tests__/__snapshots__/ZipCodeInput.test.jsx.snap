// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipCodeInput renders 1`] = `
<TextInput
  changePasswordVisibilityStatus={[Function]}
  className=""
  dark={true}
  disabled={false}
  errorMessage=""
  id="input-zip-code"
  inputDidChange={false}
  isValid={true}
  label=""
  name=""
  onBlur={null}
  onChange={[Function]}
  placeholder="Zip Code"
  shouldShowVisibility={false}
  type="text"
  value=""
/>
`;
